clang/TranslationUnit.d

```diff
@@ -113,6 +115,139 @@ struct TranslationUnit
         return Cursor(r);
     }
 
+    SourceLocation location (uint offset)
+    {
+        CXFile file = clang_getFile(cx, spelling.toStringz);
+        return SourceLocation(clang_getLocationForOffset(cx, file, offset));
+    }
+
+    SourceLocation location (string path, uint offset)
+    {
+        CXFile file = clang_getFile(cx, path.toStringz);
+        return SourceLocation(clang_getLocationForOffset(cx, file, offset));
+    }
+
+    SourceRange extent (uint startOffset, uint endOffset)
+    {
+        CXFile file = clang_getFile(cx, spelling.toStringz);
+        auto start = clang_getLocationForOffset(cx, file, startOffset);
+        auto end = clang_getLocationForOffset(cx, file, endOffset);
+        return SourceRange(clang_getRange(start, end));
+    }
+
+    package SourceLocation[] includeLocationsImpl(Range)(Range cursors)
```

Comment:
It is used in the next function that returns a comparator that allows to compare locations of cursors globally, across different include files. It is needed because libclang report all preprocessor cursors at the beginning and I needed to reorder them to correctly translate them.